equipment_handler/src/adcs_handler.c

```diff
@@ -403,10 +534,32 @@ ADCS_returnState ADCS_initiate_download_burst(uint8_t msg_length, bool ignore_ho
     return adcs_telecommand(command, 3);
 }
 
-void ADCS_receive_download_burst(uint8_t *hole_map, uint8_t *image_bytes, uint16_t length_bytes) {
+void ADCS_receive_download_burst(uint8_t *hole_map, int32_t file_des, uint16_t length_bytes) {
 #if defined(USE_UART)
+    ADCS_returnState err = ADCS_OK;
     for(int i = 0; i < length_bytes/20; i++) {
-        receieve_uart_packet(hole_map, image_bytes);
+        uint8_t pckt[20] = {0};
+        uint16_t pckt_counter;
+        err = receive_file_download_uart_packet(pckt, &pckt_counter);
+
+        if(err = ADCS_UART_FAILED){
+            // End of file
+            break;
+        }else if(pckt_counter != i){
+            // Missed a packet (or more) somewhere - fill with zeroes
+            for(int j = 0; j < (pckt_counter - i); j++){
+                write_pckt_to_file(file_des, "00000000000000000000", ADCS_UART_FILE_DOWNLOAD_PKT_DATA_LEN);
+            }
+        }else{
+            // Packet received. Write to file
+            write_pckt_to_file(file_des, pckt, ADCS_UART_FILE_DOWNLOAD_PKT_DATA_LEN);
+
+            // Fill hole map with a 1 indicating packet received
+            // Note hole map is stored little-endian
+            uint8_t hole_map_byte_index = pckt_counter / 8;
```

Comment:
Maybe it would be best to use a macro here for 8?

Reply:
I've removed this commit from the merge. Added it accidentally the other day. But I'll implement this in the hardware_testing_dev branch